packages/expo/scripts/resolveAppEntry.js

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env node
+
+// Basic node CLI script to resolve the native app entry file.
+//
+// Usage:
+//   node expo/scripts/resolveAppEntry.js path/to/project-root <platform> <format>
+//
+// Example:
+//   node expo/scripts/resolveAppEntry.js path/to/project-root/ android absolute
+//
+// Returns:
+//   The resolved entry file path.
+//
+// Limitations:
+//   Currently only supports android and ios.
+
+const { resolveEntryPoint } = require('@expo/config/paths');
+const path = require('path');
+
+const projectRoot = process.argv[1];
+const platform = process.argv[2];
+const absolute = process.argv[3] === 'absolute';
+
+if (!platform || !projectRoot) {
+  console.error(
+    'Usage: node expo/scripts/resolveAppEntry.js <projectRoot> <platform> <relative|absolute>'
+  );
+  process.exit(1);
+}
+
+const entry = resolveEntryPoint(projectRoot, { platform });
+
+if (entry) {
+  console.log(absolute ? path.resolve(entry) : path.relative(projectRoot, entry));
```

Comment:
One thing I wasn't sure about is whether `entry` is relative, absolute, or something else. If it's guaranteed to be absolute, the path.resolve() call could go away. And if it's not, we might want path.relative(projectRoot, path.resolve(entry)) in the other code path.

Reply:
I think right now it can return both since we have [pass-through conditions](https://github.com/expo/expo-cli/blob/979b6ea752370bfad8328ac5fa27389464cea4f3/packages/config/src/paths/paths.ts#L64-L69).